### ISSUE TITLE
Add new field or class to describe an entry-wide coordinate reference point

### ISSUE THREAD

**Issue by PeterC-DLS** [nexusformat/definitions#928]
This could be an physical survey point or a virtual point to anchor any NXtransformations chains. If another reference point is defined then their relative offset can be used to define coordinate frame axes.

Some experiments do not have a sample so defining an origin as the centre of a scattering volume could be problematic. It may aid the use of NeXus to describe an optical train for beamline simulations.

**Comment by yayahjb** [comment]
Good idea.  How about a new NXcoordinate_frame base class with fields for reference_point and coordinate_frame
